Progress Report March 2020 (#168)
yuzu-emu/yuzu-emu.github.io: site/content/entry/yuzu-progress-report-mar-2020/index.md

+++
date = "2020-04-19T04:00:00-03:00"
title = "Progress Report March 2020"
author = "GoldenX86"
forum = 217502
+++ 

> How are you doing in this quarantine, yuz-ers? We hope you are all staying in your homes safe and sound.

Reviewer: `yuz-ers`
sounds way too close to losers.

:P

Author: Thanks, now I can't unsee it. This wasn't me, Vortex likes it, looks like it was used before.

Today, we are bringing you a report on a few changes from the past month for both the Early Access and regular Mainline versions of yuzu.
<!--more-->

We have on our plate a couple of bug fixes, some graphical improvements, the first “performance win” for Intel GPUs, a really serious case of serendipity, and some groundwork for amazing improvements that will be coming to you in a “near-ish” future.

## Bug fixes and improvements

After some intensive debugging by [Rodrigo](https://github.com/ReinUsesLisp), an OpenGL rendering bug in ARMS (that was with us since the game started to boot), was solved by simply updating our buildbot VM from Visual Studio 2017 to the latest 2019 version.
Thanks to [DarkLordZach](https://github.com/DarkLordZach) for the change!

PSA - Please update to or install the latest [Visual C++ 2019 x64 redistributable](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads), or else you won’t be able to open yuzu, as it will complain that the library VCRUNTIME140.dll cannot be found.

{{< single-title-imgs
    "Better, right?"
    "./03.png"
    "./04.png"
  >}}

> perillamint and [bunnei](https://github.com/bunnei) stubbed (this means to send a fake `everything’s OK` signal so the game can continue) some VR related services.

Reviewer 2: Yes this is correct

These changes are needed for version 1.6.0 of The Legend of Zelda: Breath of the Wild, 1.3.0 of Super Mario Odyssey, and versions 3.1.0 and above of Super Smash Bros. Ultimate.

Commercial games aren't everything.

> [Rodrigo](https://github.com/ReinUsesLisp) decided to give homebrew lovers and old-school gamers something to play with, by implementing some attributes from the old OpenGL 1.X and 2.X days.

Reviewer 3: Remove the comma after `with`

It’s now possible to run `Xash3D`, so you can frag in Counter-Strike, or follow G-Man’s orders in Half-Life!
Currently, this needs to be manually built with newer dependencies, but that should be solved for good in the near future.

> Margen67 and [Morph](https://github.com/Morph1984) wanted to improve the graphics for games and so they added Anisotropic Filtering (AF) to yuzu.

Reviewer 3: `Margen67` should be linked to their github account. Is this the correct one? https://github.com/Margen67

> With this feature in place, distant textures now look much sharper, improving game quality even over what the original Switch does! 

Reviewer 3: Change `original Switch does!` to `original Switch can produce!`

> Games like Pokémon Sword, Pokémon Shield, and Fire Emblem Three Houses benefit a lot from this.

Reviewer 3: I believe the removing the period after this sentence and replacing it with a semicolon will make the paragraph more cohesive.

change to `from this; however, some other`

> Games like Pokémon Sword, Pokémon Shield, and Fire Emblem Three Houses benefit a lot from this.

Reviewer 3: Fire Emblem should have a colon after `Fire Emblem`

However, some other games don’t respond well to this new feature, one example being Astral Chain. Feel free to experiment with it!

An example in Pokémon Sword:

{{< imgs
    "./15.png| AF Off"
    "./14.png| AF 16x"
  >}}
  

> And another example in Fire Emblem Three Houses:  

Reviewer 3: Fire Emblem should have a colon after `Fire Emblem`

Putting all the games in italics, as I mentioned previously, will help with making the two colons less awkward

  
{{< imgs
    "./01.png| AF Off"
    "./02.png| AF 8x"
  >}}

Shader cache invalidations are something our users don’t like at all. 

> We don't too, but it’s all part of the process of improving yuzu to achieve better performance and accuracy.

Reviewer 3: Replacing `too` with `either` improves the flow of the sentence

For example, one of our recent invalidations was needed to fix the reflections in Luigi’s Mansion 3, making certain parts of the game playable.

> Because dancing is important, [namkazt](https://github.com/namkazt) and [Rodrigo](https://github.com/ReinUsesLisp) have been fighting with the code, allowing us to enjoy 3D graphics in Hatsune Miku: Project DIVA MEGA39's, and as collateral, fixed some rendering issues in Bayonetta 1 and Super Smash Bros. Ultimate.

Reviewer 3: Just a suggestion: maybe say "because the art of dancing is important" to be more eloquent?

{{< imgs
    "./07.png| Miku Miku Dance!"
  >}}

{{< single-title-imgs
    "Super Smash Bros. Ultimate's menu (Before & After)"
    "./08.png"
    "./09.png"
  >}}

[Kewlan](https://github.com/Kewlan), a regular in our Discord Server, submitted a small change to our input mapping GUI, changing the position of the shoulder buttons so they no longer make anyone angry.

{{< imgs
    "./11.png| Incorrect."
    "./12.png| Much better."
  >}}

We also have a cool improvement by [Tobi](https://github.com/FearlessTobi).
The Switch is a surround sound compatible console and our implementation of downmixing removed any channel that wasn’t left or right.
By keeping the extra channels and mixing them in the left and right stereo outputs, the sound in tons of games, including Luigi’s Mansion 3, Sonic Forces, and Bayonetta, has been improved.

## A peculiar case of "Even a broken clock is right twice a day!"

> The Nintendo Switch supports `ASTC`, a type of texture format generally used in mobile hardware such as phones and tablets in order to reduce bandwidth usage (a limited resource in those devices).

Reviewer 3: comma after `hardware` and `tablets`

Since it's a heavily compressed format, decoding it via the CPU, like it is normally done on PC, is a time-consuming task.

To give an example, Astral Chain is a game that uses `ASTC` very extensively and in bizarre ways, as previous fixes and optimizations have shown.

> On NVIDIA and AMD graphics hardware, the first scene of a new game normally takes typically `17` seconds to load. The GPU has to wait for the CPU to decode all the `ASTC` textures first, thus, slowing down the whole process.

Reviewer 3: `normally` and `typically` are redundant. 

You can make the sentence either of the two:

`On NVIDIA and AMD graphics hardware, the first scene of a new game normally takes `17` seconds to load.`
or
`On NVIDIA and AMD graphics hardware, the first scene of a new game typically takes `17` seconds to load.`

> On NVIDIA and AMD graphics hardware, the first scene of a new game normally takes typically `17` seconds to load. The GPU has to wait for the CPU to decode all the `ASTC` textures first, thus, slowing down the whole process.

Reviewer 3: Change `The GPU has to wait for the CPU to decode all the `ASTC` textures first, thus, slowing down the whole process.` 
to 
`The GPU has to wait for the CPU to decode all the `ASTC` textures, and thus slowing down the whole process.`

> On NVIDIA and AMD graphics hardware, the first scene of a new start of the game normally takes up to `17` seconds to load, as the GPU has to wait for the CPU to decode all the `ASTC` textures first, slowing down the whole process.

Reviewer 4: ```suggestion
On NVIDIA and AMD graphics hardware, the first scene of a fresh start of the game takes typically up to `17` seconds to load. The GPU has to wait for the CPU to decode all the `ASTC` textures first,  thus slowing down the whole process.
```

Splitting sentence for easier reading.

Author: Not sure if fresh start is the best option here, but new start is also bad. It should reflect saying "starting a new game", I'll edit it to something like that.

Author: Gotta love that use of "thus".

Reviewer 2: Why not `the first scene at the start of the game`

> Thanks to [Rodrigo's](https://github.com/ReinUsesLisp) research, it was discovered that on any recent Intel GPU this same operation is immediate and occurs with no delay, thanks to it being the only vendor providing a native `ASTC` decoder in their hardware.

Reviewer 3: Change `thanks to it` to `due to it` as `thanks` was already used at the beginning of the sentence.

That’s right, the best way to experience Astral Chain in yuzu is with a small, and typically inferior, iGPU (like the ones that come in any desktop or notebook CPU from Intel).

On the topic of formats, implementing the `RG32` and `RGB32` vertex formats fixed a room that refused to load in The Legend of Zelda: Link’s Awakening in Vulkan.

> Great work [Rodrigo](https://github.com/ReinUsesLisp)!

Reviewer 4: ```suggestion
Great work, [Rodrigo](https://github.com/ReinUsesLisp)!
```

{{< imgs

>     "./13.png| Works!"

Reviewer 3: `It works!` sounds better

  >}}
  

> ## Paving the road for even bigger changes

Author: Undecided on this one, both don't seem to cover what I want to say.

> ## Paving the road for even bigger changes

Reviewer 3: `bigger` or `more significant` are appropriate depending on the context of the usage.

> [Rodrigo](https://github.com/ReinUsesLisp)'s initial work on Transform Feedback has concluded and is now partially working. 

Reviewer 3: Maybe encapsulate `Transform Feedback` with code tags?

> [Rodrigo](https://github.com/ReinUsesLisp)'s initial work on Transform Feedback is finished and it's partially working. 

Reviewer 4: ```suggestion
[Rodrigo](https://github.com/ReinUsesLisp)'s initial work on Transform Feedback has concluded and partially working. 
```

Author: I like it, but it misses and *is* partially working.

> This already helps games like POKKÉN Tournament DX, Donkey Kong Country: Tropical Freeze and Xenoblades Chronicles.

Reviewer 3: I would also add `has` after `This` if changing `helps` to `helped`

An optimization by [Blinkhawk](https://github.com/FernandoS27) reduced the use of RAM by 1GB.
When talking about RAM usage, less is best!

> Now, an important but also astonishing fix; one that nobody expected, not even our project leader [bunnei](https://github.com/bunnei).

Reviewer 3: Add `for` after `Now,`

While he was working on fixing what seemed to be a simple bug on Final Fantasy VIII, he also accidentally fixed Super Smash Bros. Ultimate fights!

> This case of serendipity is without a doubt a surprise, but a welcome one.

Reviewer 4: ```suggestion
This case of serendipity is, without a doubt, a surprise, but a welcome one.
```

> Now all users can brawl all they want thanks to a measly 5-line change in yuzu’s code!

Reviewer 3: Using `all` back to back reads poorly. Perhaps change the 2nd `all they want` to `to their hearts' content`

> However, keep in mind that any Super Smash Bros. Ultimate update newer than 3.1.0 will work, but will also make the characters T-pose, resulting in very funny fights and victory screens.

Reviewer 4: ```suggestion
However, keep in mind that any Super Smash Bros. Ultimate update newer than 3.1.0 will work, but will also make the characters T-pose, resulting in hilarious fights and victory screens.
```

We’re working on resolving this, but for now, stick to update 3.1.0 or older!

[bunnei](https://github.com/bunnei) fixed audio crackling in various games like Crash Team Racing and Super Mario Odyssey.

> We are slowly, but surely, improving our audio quality.

Reviewer 3: Remove the commas after `slowly` and `surely`

He also implemented initial support for `32-bit ARM` emulation. 
Nothing tangible can be tested yet, but thanks to this, it will be possible in the future to run `32-bit` games like Mario Kart 8 Deluxe.

> And last but not least, [Blinkhawk](https://github.com/FernandoS27) has been making some serious changes to the yuzu's GPU emulation, while also laying the foundation for the recent rework of yuzu's Virtual Memory Manager (codename: Project Epimetheus).

Reviewer 3: Remove the `the` in `to the yuzu's GPU emulation`

> These, in turn, will help further advance the development of Project Prometheus; one of the super-secret Projects our devs are currently working on to bring some huge changes to yuzu’s performance and hardware requirements.

Reviewer 4: ```suggestion
These, in turn, will help further advance the development of Project Prometheus; one of the super-secret Projects our devs are currently working on to bring some considerable changes to yuzu’s performance and hardware requirements.
```

Author: Ehh, it's a bit more than considerable.

> These, in turn, will help further advance the development of Project Prometheus; one of the super-secret Projects our devs are currently working on to bring some considerable changes to yuzu’s performance and hardware requirements.

Reviewer 3: Replace the `;` with a comma. The 2nd sentence isn't an independent clause.

Stay tuned to hear more about them in the future!

> That’s all for now folks, see you on the April report.

Reviewer 3: Replace `on` with `in`

&nbsp;
<h4 style="text-align:center;">
<b>Please consider supporting us on [Patreon](https://www.patreon.com/yuzuteam)!<br>

> If you would like to contribute to this project, checkout our [GitHub](https://github.com/yuzu-emu/yuzu)!</b>

Reviewer 4: ```suggestion
If you would like to contribute to this project, check out our [GitHub](https://github.com/yuzu-emu/yuzu)!</b>
```

Author: That's our default footnote, I don't want to edit it.

Reviewer 3: Liu is correct, checkout should be `check out`. Can't believe we never caught that lol
